### PR TITLE
EVG-1881 Add integration tests for new agent

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

**Comment by bsamek** [comment]
The yaml file is a straight copy from the old agent, so you can ignore it in the review.

File: proto/task.go [hunk 1]
```diff
@@ -18,9 +19,14 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext, complete chan<- st
 	}
 	a.checkIn(ctx, tc, initialSetupCommand, initialSetupTimeout, idleTimeout)
 
-	tc.logger.Task().Infof("Task logger initialized (agent revision: %s).", evergreen.BuildRevision)
-	tc.logger.Execution().Info("Execution logger initialized.")
-	tc.logger.System().Info("System logger initialized.")
+	if ctx.Err() != nil {
+		grip.Info("task canceled")
```
Comment (john-m-liu) [inline proto/task.go]: Should this log the error instead? You can wrap it with "task canceled"

File: proto/task.go [hunk 2]
```diff
@@ -30,7 +36,7 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext, complete chan<- st
 	}
 
 	if ctx.Err() != nil {
-		tc.logger.Task().Info("task canceled")
+		grip.Info("task canceled")
```
Comment (john-m-liu) [inline proto/task.go]: Same here - should maybe log the error instead

File: proto/agent_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	suite.Run(t, new(AgentIntegrationSuite))
+}
+
+func (s *AgentIntegrationSuite) SetupSuite() {
+	s.testConfig = testutil.TestConfig()
+	testutil.ConfigureIntegrationTest(s.T(), s.testConfig, "AgentIntegrationSuite")
+	dbutil.SetGlobalSessionProvider(dbutil.SessionFactoryFromConfig(s.testConfig))
+}
+
+func (s *AgentIntegrationSuite) SetupTest() {
```
Comment (john-m-liu) [inline proto/agent_integration_test.go]: I don't think this needs to be here if it's empty

File: proto/agent_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	s.tc = &taskContext{
+		task: client.TaskData{
+			ID:     s.modelData.Task.Id,
+			Secret: s.modelData.Task.Secret,
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	err = s.a.startNextTask(ctx, s.tc)
```
Comment (john-m-liu) [inline proto/agent_integration_test.go]: Does it make sense to check the status of the task after it's run here?

**Review by john-m-liu — CHANGES_REQUESTED** [review]

File: testutil/testing.go [hunk 5]
```diff
@@ -12,13 +12,14 @@ import (
 )
 
 const (
-	envAll      = "EVERGREEN_ALL"
+	// EnvAll is an environment variable to set in order to run integration tests.
+	EnvAll      = "EVERGREEN_ALL"
```
Comment (bsamek) [inline testutil/testing.go]: Done.

File: proto/agent.go [hunk 6]
```diff
@@ -133,6 +131,11 @@ func (a *Agent) startNextTask(ctx context.Context, tc *taskContext) error {
 		"task_secret": tc.task.Secret,
 	})
 
+	// Defers are LIFO. We cancel all agent task threads, then any procs started by the agent, then remove the task directory.
+	defer a.removeTaskDirectory(tc)
```
Comment (bsamek) [inline proto/agent.go]: True, but we will not have created the task directory or any procs, and we will not have anything to cancel.

File: proto/task.go (hunk 1 above)
Comment (bsamek) [inline proto/task.go]: I don't think so. This error just means that the context has been canceled.

File: proto/agent_integration_test.go (hunk 3 above)
Comment (bsamek) [inline proto/agent_integration_test.go]: Done.

File: proto/agent_integration_test.go (hunk 4 above)
Comment (bsamek) [inline proto/agent_integration_test.go]: What do you mean by the task status? Do you mean in the db?

File: proto/agent_test.go [hunk 7]
```diff
@@ -109,3 +109,17 @@ func (s *AgentTestSuite) TestEndTaskResponse() {
 	s.False(detail.TimedOut)
 	s.Equal(evergreen.TaskFailed, detail.Status)
 }
+
+func (s *AgentTestSuite) TestAbort() {
+	s.mockCommunicator.HeartbeatShouldAbort = true
+	err := s.a.startNextTask(context.Background(), s.tc)
+	s.NoError(err)
+	s.Equal("failed", s.mockCommunicator.EndTaskResult.Detail.Status)
```
Comment (bsamek) [inline proto/agent_test.go]: Done.

File: proto/task.go (hunk 2 above)
Comment (bsamek) [inline proto/task.go]: Same thing here. There's no error, just a context cancelation.

**Review by john-m-liu — APPROVED** [review]

**Pull request merged**